bridge/slack/helpers.go

```diff
@@ -63,15 +63,18 @@ func (b *Bslack) getChannelBy(lookupKey string, lookupMap map[string]*slack.Chan
 
 const minimumRefreshInterval = 10 * time.Second
 
-func (b *Bslack) populateUsers() {
+func (b *Bslack) populateUsers(wait bool) {
 	b.refreshMutex.Lock()
-	if time.Now().Before(b.earliestUserRefresh) || b.refreshInProgress {
+	if !wait && (time.Now().Before(b.earliestUserRefresh) || b.refreshInProgress) {
 		b.Log.Debugf("Not refreshing user list as it was done less than %v ago.",
 			minimumRefreshInterval)
 		b.refreshMutex.Unlock()
 
 		return
 	}
+	for b.refreshInProgress {
+		time.Sleep(time.Second)
```

Comment:
You can actually determine the right duration based on current time and `b.earliestUserRefresh`.

Also you will likely want to rewrite the logic somewhat here as you'll need to release the lock while you are sleeping. And once you are finished you'll need to recheck on whether you have passed the `earliestUserRefresh` time (as there's always the possibility of a race-condition.

Reply:
Yes, but I want to do it earlier then the `b.earliestUserRefresh` that's why I don't use it here :-)
Why should I release the lock while sleeping? It may block here, as it'll only happen on startup where we want the blocking.

Reply:
I'm going to merge this now so I can release a v1.12.3 which has a couple of slack bugfixes. We can rewrite this later in a safer way.